sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -1278,6 +1299,13 @@ private HandlesSplits.SplitResult trySplitForWindowObservingTruncateRestriction(
       if (currentWindow == null) {
         return null;
       }
+      // We are requesting a checkpoint but have not yet progressed on the restriction, skip
+      // request.
+      if (fractionOfRemainder == 0
+          && currentTrackerClaimed != null
```

Comment:
Given it is not marked `volatile` those could be out of date, right? Based on this and your use of `lazySet` that seems to align with your intent, but I don't have a holistic understanding of this code.